packages/client/tests/functional/issues/9007/tests.ts

```diff
@@ -0,0 +1,45 @@
+import fs from 'fs'
+import path from 'path'
+
+import testMatrix from './_matrix'
+// @ts-ignore
+import type { PrismaClient } from './node_modules/@prisma/client'
+
+declare let prisma: PrismaClient
+
+// https://github.com/prisma/prisma/issues/9007
+testMatrix.setupTestSuite(
+  () => {
+    test('should throw an error if using contains filter on uuid type', async () => {
+      await prisma.user.create({ data: {} })
+
+      await expect(() =>
+        prisma.user.findMany({
+          where: {
+            // @ts-expect-error
+            uuid: { contains: 'foo-bar' },
+          },
+        }),
+      ).rejects.toThrowError()
```

Comment:
**This does not throw an error.** 

We should be throwing a runtime validation error here.